### PR TITLE
Update Docker instructions to be more automated-build friendly

### PR DESCRIPTION
**Pull request by turt2live**
This PR also assumes that the image will end up on docker.io (as an automated build).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,21 @@
+# Build go-neb
+FROM golang:1.8-alpine as builder
```
Comment (jcgruenhage) [inline Dockerfile]: why 1.8-alpine and not 1.10-alpine?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,10 @@
+vendor/pkg
+vendor/src
+pkg
+hooks
+bin
+
+# Other stuff we just don't need
+.gitignore
+.git
+.travis.yml
```
Comment (jcgruenhage) [inline .dockerignore]: iirc the .dockerignore file is only important for not having to hand a lot of data to the docker daemon over http. I agree that .git and some of the directories up there should definitely be in there, but ignoring .gitignore and .travis.yml seems a little random

File: README.md [hunk 3]
```diff
@@ -225,7 +225,12 @@ sensible-browser http://localhost:6060/pkg/github.com/matrix-org/go-neb
 
 ## Docker image
 
-There is a `Dockerfile` in the root of the repository and a `build-docker-image.sh` script that uses an alpine-based golang container to build `go-neb` (note that this will overwrite host-built binaries in `pkg/` and `bin/`) and then builds the docker image using that binary.
+To get started quickly, use the image from docker.io:
+```
+docker run -v /path/to/data:/data -e "BASE_URL=http://your.public.url:4050" matrixdotorg/go-neb
+```
+
+If you'd prefer to build the file yourself, clone this repository and build the `Dockerfile`.
```
Comment (jcgruenhage) [inline README.md]: A lot better than before, thanks!

File: .dockerignore (hunk 2 above)
Comment (turt2live) [reply .dockerignore]: They were picked randomly, so I'll remove them

File: Dockerfile (hunk 1 above)
Comment (turt2live) [reply Dockerfile]: Copy/pasta from the old script - will update

**Pull request merged**